planner/core/rule_build_key_info.go

```diff
@@ -222,6 +223,7 @@ func checkIndexCanBeKey(idx *model.IndexInfo, columns []*model.ColumnInfo, schem
 		for i, col := range columns {
 			if idxCol.Name.L == col.Name.L {
 				if !mysql.HasNotNullFlag(col.Flag) {
+					uniqueKey = append(uniqueKey, schema.Columns[i])
```

Comment:
This logic is wrong. For example, if the unique index is `key(a,b,c)`, such that column `a`, `b`, `c` may contain `null` respectively, the returned `uniqueKey` only contains column `a`.

Reply:
done